rs/nns/cmc/src/main.rs

```diff
@@ -139,6 +139,9 @@ pub enum NotificationStatus {
     NotifiedCreateCanister(Result<CanisterId, NotifyError>),
     /// The cached result of a completed cycles mint.
     NotifiedMint(NotifyMintCyclesResult),
+    /// The transaction did not have a supported memo (or icrc1_memo).
+    /// Therefore, we decided to send the ICP back to its source (minus fee).
+    AutomaticallyRefunded(Option<BlockIndex>),
```

Comment:
Nit. 

`AutomaticallyRefunded` seems like the consequence of not having a supported memo.

Should the status be called `MissingMemo`?

```suggestion
    MissingMemo(Option<BlockIndex>),
```

Reply:
Could you specify in the doc string what the argument `Option<BlockIndex>` represents?

Reply:
It's not "MissingMemo" but rather "BadMemo".

I agree that "AutomaticallyRefunded" is the consequence, and therefore does not tell the user how to fix their next request.  If it told the user what the problem was, it would be fine.